### PR TITLE
Feature/Fix: [GS] Add RoadTiles and GetOneWayInfo endpoints

### PR DESCRIPTION
**Pull request by memmpf**
## Motivation / Problem

Currently there isn't a way for GameScript to get information about road direction, shape or if it is one-way. Same kind of functionality already exists for railways.

## Description

This pull request adds two new endpoints to script_road: GetRoadTiles and GetOneWayInfo. Also corresponding enums are added to describe the shape of the responses of these endpoints. There was also a bug in calling GetAnyRoadBits which casted the road_tram_type incorrectly.

## Limitations

No known limitations.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/script/api/script_road.hpp [hunk 1]
```diff
@@ -85,6 +85,42 @@ class ScriptRoad : public ScriptObject {
 		BT_TRUCK_STOP, ///< Build a truck stop
 	};
 
+	/**
+	 * A bitmap of all the possible road tiles and combinations.
+	 */
+	enum RoadTiles {
```
Comment (JGRennison) [inline src/script/api/script_road.hpp:91]: This isn't about tiles at all. It's probably be better to rename it something like RoadBits or RoadPieces or similar?

File: src/script/api/script_road.hpp [hunk 2]
```diff
@@ -85,6 +85,42 @@ class ScriptRoad : public ScriptObject {
...
+	 */
+	enum OneWayInfo {
+		/* Note: Follows the same pattern as RoadCachedOneWayState. */
+		ONEWAY_NONE = 0,							///< Not a one-way road.
+		ONEWAY_NON_JUNCTION_A,				///< One-way road, no junctions, south-west to north-east + north-east to south-west
+		ONEWAY_NON_JUNCTION_B,				///< One-way road, no junctions, north-west to south-east + south-east to north-west
+		ONEWAY_NO_ACCESS,							///< Road is disallowed in both directions
+		ONEWAY_SIDE_JUNCTION,					///< One-way road junction, can exit
+		ONEWAY_SIDE_JUNCTION_NO_EXIT,	///< One-way road junction, can't exit
+	};
```
Comment (JGRennison) [inline src/script/api/script_road.hpp:122]: RoadCachedOneWayState is an internal structure which is subject to change.
It'd probably be better to use DisallowedRoadDirections/GetDisallowedRoadDirections.

**Comment by memmpf** [comment]
Now the naming of GetRoadTiles is changed to GetRoadPieces, and corresponding enum keys also follow the same convention. The GetOneWayInfo was changed to use GetDisallowedRoadDirections and enum was changed to represent direction of the oneway road.

**Comment by JGRennison** [comment]
Thanks for this

**Pull request merged**